syntax = "proto3";

package llm;

service LLMService {
    rpc Analyze(AnalyzeRequest) returns (AnalyzeResponse);
}

message AnalyzeRequest {
    string user_query = 1;
    string stats_json = 2;
    string context = 3;
}

message AnalyzeResponse {
    string answer_text = 1;
    string reasoning = 2;
    string structured = 3;
}
